src/views/CoCurricularTranscript/index.js

```diff
@@ -170,36 +186,33 @@ export default class Transcript extends Component {
 
     // sort activities by ActivityCode
     while (activities.length > 0) {
-      // shift removes and returns the first item in the array
-      let curAct = activities.shift();
-      let sessions = [curAct.SessionCode];
-
-      // Get the session codes for all activities matching the current activity's code,
-      // knowing activities will be sorted by session and Activity Code.
-      // Continue looking at activities while the activity at array front has the
-      // same activityCode as the current activity
-      while (activities.length > 0 && activities[0].ActivityCode === curAct.ActivityCode) {
-        sessions.push(activities.shift().SessionCode);
-      }
+      let curAct = activities[0];
 
-      // Translate the sessions array into a duration:
+      // keep track of the activity code which will be used to identify all activities of the same
+      // code so they can be grouped into one activity component
+      let curActCode = curAct.ActivityCode;
 
-      // Pop first session code from array and split into months and years
-      let sess = sessions.shift();
+      // Pop first session code from array and split into months and years, which are saved as
+      // the initial start and end dates
+      // (shift is like pop, removes and returns the first item in the array)
+      let sess = activities.shift().SessionCode;
```

Comment:
Why shift activities here instead of shifting it for curAct above and then taking sessioncCode from curAct?